src/Disks/IO/WriteBufferFromAzureBlobStorage.cpp

```diff
@@ -77,62 +89,77 @@ void WriteBufferFromAzureBlobStorage::execWithRetry(std::function<void()> func,
 
 void WriteBufferFromAzureBlobStorage::finalizeImpl()
 {
+    auto block_blob_client = blob_container_client->GetBlockBlobClient(blob_path);
+
+    if (buffer_allocation_policy->getBufferNumber() == 1)
+    {
+        size_t data_size = size_t(position() - memory.data());
+        if (data_size <= max_single_part_upload_size)
```

Comment:
And what if `(data_size > max_single_part_upload_size)` here?

Reply:
added a comment , it will go to multipart upload